subresource-integrity/integrity-policy/tentative/signature.https.html

```diff
@@ -0,0 +1,139 @@
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/common/dispatcher/dispatcher.js"></script>
+<script src="/common/utils.js"></script>
+<script src="/common/get-host-info.sub.js"></script>
+<script src="/reporting/resources/report-helper.js"></script>
+
+<body>
+<script>
+  const {ORIGIN} = get_host_info();
+  const getAbsoluteUrl = url => {
+    return new URL(url, window.location.href).href;
+  }
+
+  const check_report = async (reporting_endpoint, reporting_uuid, iframe_url, url, report_only) => {
+    const reports = await pollReports(reporting_endpoint, reporting_uuid);
+    const abs_iframe_url = getAbsoluteUrl(iframe_url);
+    checkReportExists(reports, 'integrity-violation', abs_iframe_url);
+    const abs_blocked_url = getAbsoluteUrl(url);
+    const report = getReport(reports, 'integrity-violation', abs_iframe_url, abs_blocked_url);
+    assert_not_equals(report, null);
+    assert_equals(report.body.documentURL, abs_iframe_url);
+    assert_equals(report.body.blockedURL, abs_blocked_url);
+    assert_equals(report.body.destination, "script");
+    assert_equals(report.body.reportOnly, report_only);
+  };
+  const blob = new Blob([`window.ran=true;`],
+                        { type: 'application/javascript' });
+
+  const blob_url = URL.createObjectURL(blob);
+
+  // Generated using https://sha2.it/ed25519.html (In Chrome Canary, with Experimental Web Platform Features enabled)
+  const signature = encodeURIComponent(
+    'header(Unencoded-Digest, sha-384=:tqyFpeo21WFM8HDeUtLqH20GUq\/q3D1R6mqTzW3RtyTZ3dAYZJhC1wUcnkgOE2ak:)' +
+    '|header(Signature-Input, signature=\\("unencoded-digest";sf\\); keyid="JrQLj5P\/89iXES9+vFgrIy29clF9CC\/oPPsw3c5D0bs="; tag="sri")' +
+    '|header(Signature, signature=:qM19uLskHm2TQG5LJcH/hY0n0BWWzYOJztVWYlwk0cZb3u0JdgUMre1J4Jn8Tma0x2u5/kPBfbXRMbB+X+vTBw==:)');
+
+  const test_cases = [
+    {
+      description: "Ensure that a script with signature integrity runs",
+      url: "/content-security-policy/resources/ran.js?pipe=" + signature,
+      cross_origin: true,
+      integrity: "ed25519-JrQLj5P/89iXES9+vFgrIy29clF9CC/oPPsw3c5D0bs=",
+      policy_violation: false,
+      block: true,
+      endpoints: true,
+      expected: {blocked: "", ran: true },
+    }
+  ];
+  test_cases.map(test_case => {
+    promise_test(async () => {
+      const REMOTE_EXECUTOR =
+        `/common/dispatcher/remote-executor.html?pipe=`;
+      const iframe_uuid = token();
+
+      const params = new URLSearchParams(location.search);
+      if (params.get('type') === "report") {
+        if (test_case.expected.blocked) {
+          return;
+        }
+        header_name += "-Report-Only";
+      }
+      const reporting_uuid_1 = token();
+      const reporting_uuid_2 = token();
+      const reporting_uuid_3 = token();
+      const reporting_endpoint = `${ORIGIN}/reporting/resources/report.py`;
+      let header = "";
+      if (test_case.block) {
```

Comment:
This is actually just always true. Would it make sense to get rid of the test_cases and foreach?

As usual, I also personally believe that the whole remote-executor logic makes this unreadable, and that it would be nicer to just have a file with a corresponding header file and run this test in the top level.

My understanding is that the remote-executor logic was implemented for tests where different origins are interacting and we need to collect information from different origins for test assertions - which is not otherwise possible to do. In all other cases, I would suggest avoiding this.

Reply:
I'll simplify. Remote-executor is significantly simpler for tests that involve reporting, as you can coordinate the UUID between the test logic and the headers. Otherwise, you have to pass the UUID using Server-Timing, which is feasible, but hacky.

Reply:
OK, another reason why ExecutorContext is easier - without it and with the Integrity-Policy headers in place, the testharness scripts fail to load.. I can add integrity hashes to them, but that would break every time those scripts change..

Reply:
ok, fair enough.

Some nits, but otherwise LGTM.

Reply:
Thanks!